### PR TITLE
Tutorial explaining historical parameters

### PR DESCRIPTION
**Pull request by OFR-IIASA**
This PR extends the Westeros tutorial set to explain historical parameters, specifically `historical_new_capacity`.

## How to review

- Read the diff and note that the CI checks all pass.
- Run a specific the new tutorial -> check that it works and that the explanations are understandable.

## PR checklist

- [x] Continuous integration checks all ✅

- [x] Add or expand tests; coverage checks both ✅
- [x] Add, expand, or update documentation.
- [x] Update release notes.

### CODE REVIEW

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:176]: a minor issue related to capitalizing the words in titles: "Defining generic functions"

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:216]: Maybe for the section numbers something like 2.1 or II.1 would be better?

**Review by behnam-zakeri — CHANGES_REQUESTED** [review]
@OFR-IIASA thanks a lot for this informative and well formatted tutorial. It is very useful and the text helps to understand different steps. I suggest not using efficiency and capacity factor of 100% for thermal power plants, because we know that these are just simple tutorials, but sometimes such assumptions are confusing for users. Otherwise, I believe this can be merged.

**Comment by OFR-IIASA** [comment]
> 
> 
> @OFR-IIASA thanks a lot for this informative and well formatted tutorial. It is very useful and the text helps to understand different steps. I suggest not using efficiency and capacity factor of 100% for thermal power plants, because we know that these are just simple tutorials, but sometimes such assumptions are confusing for users. Otherwise, I believe this can be merged.

@Behnam, Thanks for the review.  While in general I agree that it is not the best practice to use efficiency and capacity factors parameters with 100%, I think that this assumption is buried within the baseline tutorial, hence I wanted to avoid making too many changes in this tutorial. We may want to spend the time and make this uniform across all tutorials, i.e. adjust the capacity and efficiency, but then I would suggest a separate PR.

**Review by behnam-zakeri — APPROVED** [review]

**Comment by behnam-zakeri** [comment]
Thanks, Oliver. I agree with you that these values should be corrected in all tutorials. So, for this once, all looks good and I suggest merging this.

**Comment by OFR-IIASA** [comment]
@khaeru and @LauWien the PR is ready to merge if there are no further changes requested from your side.

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:138]:
```suggestion
    "After 2030, there is no continuation of policies assumed, hence the model will return to a \"baseline\" trajectory (see solid blue line *\\\"Baseline\\\"* in the panel a. of the figure below).\n",
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:139]:
```suggestion
    "This scenario will then serve as a basis for further analysis - e.g. scenario b) (see red dashed line *\\\"NDC-mitigation\\\"* in panel b. in the figure below).\n",
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:141]:
```suggestion
    "The historical time-periods (see the extension of the grey block in panel b. in the figure below) are therefore shifted.\n",
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:144]:
```suggestion
    "In the process of cloning the scenario, for the example below, all time-periods prior to 2035 will be set to \"historical\" time-periods and its solutions will be transferred to `historical_activity` and `historical_new_capacity`.\n",
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:207]:
```suggestion
    "  - We will remove the growth constraint for `coal_ppl`.\n",
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:310]:
```suggestion
    "demand_of_firstmodelyear= float(scenario.par('demand', filters={'year': scenario.firstmodelyear})['value'])"\n",
    "print(f\"The demand of the `firstmodelyear` is {demand_of_firstmodelyear} GWa.\")
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:324]:
```suggestion
    "demand_of_firstmodelyear/= grid_eff"
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:335]:
```suggestion
    "historical_new_capacity = demand_of_firstmodelyear / duration_period"
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:337]:
```suggestion
  },
{
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each year within the historical 10 year period, a 11,11 GW `coal_ppl` must be added to meet the demand of the `firstmodelyear`. Therefore, the value of `historical_new_capacity` is calculated to `historical_new_capacity = demand_of_firstmodelyear / duration_period`."
   ]
  },
```

**Review by LauWien — CHANGES_REQUESTED** [review]
That's a really helpful tutorial! Thanks for that - looks good :)
Following a few suggestions to make it maybe even more understandable. 
Maybe more general - what also help for a deeper understanding to print out even more of the values and dfs. That was positive feedback in the MESSAGEix workshop when @behnam-zakeri did that.

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:310]:
```suggestion
    "demand_of_firstmodelyear= float(scenario.par('demand', filters={'year': scenario.firstmodelyear})['value'])\n",
```

File: tutorial/westeros/westeros_historical_new_capacity.ipynb
Comment (LauWien) [inline tutorial/westeros/westeros_historical_new_capacity.ipynb:311]:
```suggestion
    "print(f\"The demand of the `firstmodelyear` is {demand_of_firstmodelyear} GWa.\")"
```

**Review by LauWien — COMMENTED** [review]
Now it should work, sorry for that :)

**Review by LauWien — APPROVED** [review]
Looks great! Thank you very much :)
I guess as soon as the tests succeed we can squash and merge it. @khaeru what do you think?

File: RELEASE_NOTES.rst
Comment (khaeru) [inline RELEASE_NOTES.rst:20]: This single added line should appear next to the 3 existing lines under "Next release / all changes"; there should be no other additions to this file. Specific version numbers are **only** added to this file when a release is made.

**Review by khaeru — CHANGES_REQUESTED** [review]
See comment on RELEASE_NOTES.rst.

File: RELEASE_NOTES.rst
Comment (OFR-IIASA) [reply RELEASE_NOTES.rst]: Done

**Comment by OFR-IIASA** [comment]
@khaeru I update the release notes. It seems that a certain test to prone to failure, but seems unrelated to this PR.
`FAILED message_ix/tests/test_reporting.py::test_reporter_from_westeros`

**Comment by khaeru** [comment]
> It seems that a certain test to prone to failure, but seems unrelated to this PR.
> `FAILED message_ix/tests/test_reporting.py::test_reporter_from_westeros`

@LauWien identified #498, fixed in #499, and will rebase this branch to ensure that picking up the change allows tests to pass.

**Review by khaeru — APPROVED** [review]

**Pull request merged**
